homeassistant/components/mqtt/__init__.py

```diff
@@ -711,7 +715,10 @@ def async_remove() -> None:
             if any(other.topic == topic for other in self.subscriptions):
                 # Other subscriptions on topic remaining - don't unsubscribe.
                 return
-            self.hass.async_create_task(self._async_unsubscribe(topic))
+
+            # Only unsubscribe if currently connected.
+            if self.connected:
```

Comment:
Should this be checked in other places too?

Reply:
Maybe? I don't want to blow up this PR but it wouldn't hurt to check and raise errors/ignore if applicable.